File: spring-cloud-contract-producer/src/main/java/com/baeldung/spring/cloud/springcloudcontractproducer/controller/EvenOddController.java
package com.baeldung.spring.cloud.springcloudcontractproducer.controller;


import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import org.springframework.web.bind.annotation.*;

@RestController
public class EvenOddController {
    @Autowired
    Calculator calculator;

    @GetMapping("/validate/prime-number")
    public String isNumberPrime(@RequestParam("number") Integer number) {
        return calculator.isNumberPrime(number) ? "Even" : "Odd";
    }
}

File: spring-cloud-contract-producer/src/main/java/com/baeldung/spring/cloud/springcloudcontractproducer/controller/EvenOddController.kt
package com.baeldung.spring.cloud.springcloudcontractproducer.controller

import org.springframework.beans.factory.annotation.*
import org.springframework.web.bind.annotation.*

@RestController
class EvenOddController {
    @Autowired
    private var calculator: Calculator? = null
    @GetMapping("/validate/prime-number")
    fun isNumberPrime(@RequestParam("number") number: Int): String {
        return if (calculator!!.isNumberPrime(number)) "Even" else "Odd"
    }

    @GetMapping("/echo")
    fun isNumberPrime(@RequestParam("message") message: String): String {
        return message
    }
}
